Automattic/newspack-popups
# feat: reorganize editor sidebar and add notice when test mode is enabled

### All Submissions:






### Changes proposed in this Pull Request:



Reorganizes the editor sidebar according to design mockups by @thomasguillot 

Closes #97.

@thomasguillot, I extrapolated from your mockups in a few ways:

1. I replaced all instances of the word "popup" with "campaign" to follow the updated branding.
2. When showing the "test mode" warning/notice, I couldn't figure out how to format the text as designed. The component only takes plain, unformatted strings as content, so this is the closest I could get:

<img width="786" alt="Screen Shot 2020-11-10 at 12 34 18 PM" src="https://user-images.githubusercontent.com/2230142/98724578-2f6e1a00-2351-11eb-9515-cca3b950eb72.png">

3. I updated some help text for certain options, and added some where I felt it was missing. For example:

<img width="270" alt="Screen Shot 2020-11-10 at 12 24 39 PM" src="https://user-images.githubusercontent.com/2230142/98724355-e0c08000-2350-11eb-8147-a3550444dc9a.png">

4. I moved the new "Segment" option to the Frequency panel, since it felt like the most appropriate place. Let me know if you disagree:

<img width="269" alt="Screen Shot 2020-11-10 at 12 48 41 PM" src="https://user-images.githubusercontent.com/2230142/98725967-136b7800-2353-11eb-8753-1e9787b8056c.png">

5. This is no longer applicable, but I'm keeping the examples here for reference. ~Rather than simply showing or hiding options based on whether the current campaign is inline vs. overlay, we now show disabled controls when an option is not applicable. I felt this was less confusing since the options are now split between various panels which may or may not be open at any one time. For example, when the "inline" option is enabled in the Campaign Settings panel, we show this for the "sitewide default" option in the Status & Visibility panel:~

<img width="273" alt="Screen Shot 2020-11-10 at 12 41 37 PM" src="https://user-images.githubusercontent.com/2230142/98725268-2762aa00-2352-11eb-93e9-54ef172dd34d.png">

~and this when the "inline" option is disabled:~

<img width="270" alt="Screen Shot 2020-11-10 at 12 41 42 PM" src="https://user-images.githubusercontent.com/2230142/98725282-2b8ec780-2352-11eb-9afc-726efcf8ed21.png">

~Likewise for the color options, if the campaign is inline, we show disabled controls for the non-applicable options:~

<img width="268" alt="Screen Shot 2020-11-10 at 12 41 55 PM" src="https://user-images.githubusercontent.com/2230142/98725303-30ec1200-2352-11eb-9450-31e09502d6f1.png">

~For the placement/trigger settings, we still show and hide those based on the inline setting, since they're right underneath that setting (but we could do the disabled thing here, too, for consistency):~

<img width="267" alt="Screen Shot 2020-11-10 at 12 21 56 PM" src="https://user-images.githubusercontent.com/2230142/98724872-968bce80-2351-11eb-8d68-d9e7ed74232d.png">
<img width="270" alt="Screen Shot 2020-11-10 at 12 24 39 PM" src="https://user-images.githubusercontent.com/2230142/98724875-97246500-2351-11eb-898f-e59ec9b0e122.png">

~Let me know if you disagree with this approach, and I'll be happy to refactor as you prefer.~

### How to test the changes in this Pull Request:

1. Check out this branch and run `npm run build` to compile.
2. Edit a campaign or create a new one.
3. Verify that the layout of all sidebar panels and options are as described in #97.
4. Verify that the options are shown/hidden according to whether the "inline" option is enabled or disabled.
5. Verify that the options still function as before—there should be no changes in behavior on the front-end.

### Other information:
